samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/servlet/delivery/UploadAudioMediaServlet.java

```diff
@@ -444,7 +454,21 @@ private String saveMedia(int attemptsRemaining, String mimeType, String agent,
     catch(Exception e){
       log.warn(e.getMessage());
     }
-    return mediaId.toString();
+    JsonObject json = new JsonObject();
+    json.addProperty("mediaId", mediaId);
+    json.addProperty("duration", mediaData.getDuration());
+    json.addProperty("createdDate", convertDate(mediaData.getCreatedDate(), rb.getString("delivery_date_format")));
+    return json;
+  }
+
+  private String convertDate(Date time, String dateFormat){ 
```

Comment:
Don't use util methods here, replace by this method:

https://github.com/sakaiproject/sakai/blob/master/kernel/kernel-util/src/main/java/org/sakaiproject/util/DateFormatterUtil.java#L96

Locale can be retrieved using rb.getLocale();